src/sleep/sleep.c

```diff
@@ -87,6 +88,15 @@ static int write_efi_hibernate_location(const HibernationDevice *hibernation_dev
                 return log_full_errno(log_level, r, "Failed to parse ID_FS_UUID '%s' for device '%s': %m",
                                       uuid_str, hibernation_device->path);
 
+        _cleanup_(sd_device_unrefp) sd_device *origin = NULL;
+        r = block_device_get_originating(device, &origin);
+        if (r == -ENOENT)
+                r = sd_device_get_property_value(device, "ID_DISSECT_PART_DESIGNATOR", &part_designator);
+        else if (r >= 0)
+                r = sd_device_get_property_value(origin, "ID_DISSECT_PART_DESIGNATOR", &part_designator);
+        if (r < 0 && r != -ENOENT)
+                log_debug_errno(r, "Failed to get partition designator of '%s', ignoring: %m", hibernation_device->path);
```

Comment:
I think a very brief explanation what is going on here, in a comment would be good.

i.e. what ID_DISSECT_PART_DESIGNATOR is about, why we look for the originating device, and what the result is actually relevant for, i.e. what consumes it on the resume cycle